test/CMakeLists.txt

```diff
@@ -8,6 +8,23 @@ endif()
 add_executable(hello ${hello_SOURCES})
 
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    # Test compiling a file that exposes potential issues with case
+    # sensitive paths. If files are exposed with the wrong casing, compiling
+    # once will succeed, as MSVC in Wine treats paths case insensitively.
+    # But a case sensitive dependency tracker in Ninja will force unnecessary
+    # recompilation, if the files don't exist with the right casing.
+    #
+    # This is tested only with MSVC, not with clang-cl, as MSVC STL is
+    # strict with what version of Clang it supports (currently requiring
+    # Clang 15, while common repositories only provide Clang 14). The
+    # header also triggers using a form of attributes for SAL that Clang
+    # doesn't support.
+    add_library(test_headers STATIC headers.cpp)
```

Comment:
I'm not sure if it's worth adding this testcase to the cmake file; the case sensitive clang-cl test in the previous commit should give quite enough coverage here. This makes the cmake test quite a bit slower, as cl.exe seems to be doing something weird when we compile this file (due to enabling the old style attributes for sal.h).